src/event_listener/shared.rs

```diff
@@ -502,364 +614,309 @@ fn parse_string_as_work(str: String) -> WorkspaceType {
-        (
-            ParsedEventType::LayerOpened,
-            r"openlayer>>(?P<namespace>.*)",
-        ),
-        (
-            ParsedEventType::LayerClosed,
-            r"closelayer>>(?P<namespace>.*)",
-        ),
-        (
-            ParsedEventType::FloatStateChanged,
-            r"changefloatingmode>>(?P<address>.*),(?P<floatstate>[0-1])",
-        ),
-        (
-            ParsedEventType::Minimize,
-            r"minimize>>(?P<address>.*),(?P<state>[0-1])",
-        ),
-        (
-            ParsedEventType::Screencast,
-            r"screencast>>(?P<state>[0-1]),(?P<owner>[0-1])",
-        ),
-        (
-            ParsedEventType::UrgentStateChanged,
-            r"urgent>>(?P<address>.*)",
-        ),
-        (
-            ParsedEventType::WindowTitleChanged,
-            r"windowtitle>>(?P<address>.*)",
-        ),
-        (ParsedEventType::Unknown, r"(?P<Event>^[^>]*)"),
-    ].into_iter()
-    .map(|(e, r)| (
-        e,
-        match Regex::new(r) {
-            Ok(value) => value,
-            Err(e) => {
-                // I believe that panics here are fine because the chances of the library user finding them are extremely high
-                // This check does occur at runtime though...
-                eprintln!("An internal error occured in hyprland-rs while parsing regex! Please open an issue!");
-                match e {
-                    RegexError::Syntax(str) => panic!("Regex syntax error: {str}"),
-                    RegexError::CompiledTooBig(size) => {
-                        panic!("The compiled regex size is too big! ({size})")
-                    }
-                    _ => panic!("Error compiling regex: {e}"),
-                }
+/// All Hyprland events's arg count and enum variant.
+/// The first item of the tuple is a usize of the argument count
+/// This allows for easy parsing because the last arg in a Hyprland event
+/// has the ability to have extra `,`s
+pub(crate) static EVENTS: phf::Map<&'static str, (usize, ParsedEventType)> = phf::phf_map! {
+    "workspacev2" => ((2),ParsedEventType::WorkspaceChangedV2),
+    "destroyworkspacev2" => ((2),ParsedEventType::WorkspaceDeletedV2),
+    "createworkspacev2" => ((2),ParsedEventType::WorkspaceAddedV2),
+    "moveworkspacev2" => ((3),ParsedEventType::WorkspaceMovedV2),
+    "renameworkspace" => ((2),ParsedEventType::WorkspaceRename),
+    "focusedmon" => ((2),ParsedEventType::ActiveMonitorChanged),
+    "activewindow" => ((2),ParsedEventType::ActiveWindowChangedV1),
+    "activewindowv2" => ((1),ParsedEventType::ActiveWindowChangedV2),
+    "fullscreen" => ((1),ParsedEventType::FullscreenStateChanged),
+    "monitorremoved" => ((1),ParsedEventType::MonitorRemoved),
+    "monitoraddedv2" => ((3),ParsedEventType::MonitorAddedV2),
+    "openwindow" => ((4),ParsedEventType::WindowOpened),
+    "closewindow" => ((1),ParsedEventType::WindowClosed),
+    "movewindow" => ((3),ParsedEventType::WindowMovedV2),
```

Comment:
This should also be `movewindowv2`:
```suggestion
    "movewindowv2" => ((3),ParsedEventType::WindowMovedV2),
```